### ISSUE TITLE
`safe-read-string` calls an undefined `read-string*` in ClojureScript

### ISSUE THREAD

**Issue by yatesco** [Engelberg/instaparse#170]
`safe-read-string` in https://github.com/Engelberg/instaparse/blob/master/src/instaparse/cfg.cljc:

``` clojure
#?(:clj
   (let [string-reader (wrap-reader
                         (clojure.lang.LispReader$StringReader.))]
     (defn safe-read-string
       "Expects a double-quote at the end of the string"
       [s]
       (with-in-str s (string-reader *in* nil))))

   :cljs
   (defn safe-read-string [s]
     (reader/read-string* (reader/push-back-reader s) nil)))
```

`reader` is `cljs.reader` which has no `read-string*` method, only a `read-string`.

This is on version 1.9.854

This prevents anything working as the grammar can't be read.

**Comment by yatesco** [comment]
Reproducible repo: https://github.com/yatesco/instaparse-fail

**Comment by Engelberg** [comment]
Thanks for the report.  Clojurescript doesn't seem to value backwards compatibility the way Clojure does, making it difficult to keep up with Clojurescript changes.  `read-string*` was in cljs.reader until a few weeks ago.  Will need to figure out a workaround.

**Comment by yatesco** [comment]
`push-back-reader` has also gone and `(def read-string cljs.reader/read-string)` fails some of the tests

**Comment by pkpkpk** [comment]
I have a [fix](https://github.com/Engelberg/instaparse/pull/169) for this. Sorry I should've created an issue.

**Comment by Engelberg** [comment]
Thanks @pkpkpk, I saw your pull request come through, but hadn't looked at it yet as I was on vacation.  Glad to know it applies to this issue.  I will take a look at it soon.

**Comment by yatesco** [comment]
Hi @Engelberg - gentle prompt :-) - any idea when you will merge this?

**Comment by Engelberg** [comment]
I've reviewed the pull request and am ready to merge it, but had been hoping to take care of a few more changes at the same time.  Doesn't look like that is going to happen for a couple more weeks, so I'll go ahead and merge this tonight.  Thanks for the reminder.

**Comment by yatesco** [comment]
Love it - I ask you to do work and you thank me :-).

Thanks again for Instaparse!

**Comment by Engelberg** [comment]
Fixed in version 1.4.8.

Please note that Clojurescript does not currently work with Clojure 1.9-alpha20 or newer: https://dev.clojure.org/jira/browse/CLJS-2352

I ran into this Clojurescript bug when testing the pull request.
